### PR TITLE
osw converter muon update

### LINKED ISSUE #1578: ensure GNSS-R OSW converter works for Muon OSW products
## Description

the live Muon data products are now being distributed, ensure the decoder works for a both a L3 gridded and a L2 trackwise product for Ocean surface wind

## Acceptance Criteria (Definition of Done)
converter decodes Muon wind and remains backwardly compatible with the other GNSS-R wind products

## Dependencies
an input Muon wind file

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

This add the capability to decode the Muon Level-3 ocean surface wind product

## Issue(s) addressed

Resolves #1578 

## Impact

Backward compatible and additionally capable converter

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by smaticka** [comment]
@BenjaminRuston how can we test this? is there a new muon file to test convert?

**Comment by fabiolrdiniz** [comment]
Whenever testing, please remember to sync the filters of https://github.com/JCSDA-internal/skylab/blob/develop/obs/ufo/osw_muon.yaml with the ones for CYGNSS & Spire.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -2172,6 +2174,17 @@ endif()
...
                             obs.20241001T000000Z_PT15M_balloonsonde.nc4 ${IODA_CONV_COMP_TOL})
 
+  ecbuild_add_test( TARGET  test_${PROJECT_NAME}_osw_cygnss
+                    TYPE    SCRIPT
+                    ENVIRONMENT "PYTHONPATH=${IODACONV_PYTHONPATH}"
+                    COMMAND bash
+                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                            netcdf
+                            "${Python3_EXECUTABLE} ${CMAKE_BINARY_DIR}/bin/osw_2ioda.py
+                            -i testinput/cygnus.sample.20240401T000600_PT1M.l2.wind_trackgridsize25km_NOAAv1.2.nc
```
Comment (BenjaminRuston) [inline test/CMakeLists.txt]:
```suggestion
                            -i testinput/cygnss.sample.20240401T000600_PT1M.l2.wind_trackgridsize25km_NOAAv1.2.nc
```

**Comment by BenjaminRuston** [comment]
@fcvdb and added a ctest ,,,, it's for cygnss but should get something in there I sampled a file down to 70-odd points ;0)

**Review by fcvdb — APPROVED** [review]
I ran the `test_iodaconv_osw_cygnss` and it passed! Thanks.

**Comment by BenjaminRuston** [comment]
> Whenever testing, please remember to sync the filters of https://github.com/JCSDA-internal/skylab/blob/develop/obs/ufo/osw_muon.yaml with the ones for CYGNSS & Spire.

@fabiolrdiniz are you just saying we should update the skylab yaml for muon?

**Review by smaticka — APPROVED** [review]
thanks @BenjaminRuston! this worked for me when I ran `ctest -R test_iodaconv_osw_cygnss -VV`. 
Is the ctest name in the `CMakeList.txt` file `test_${PROJECT_NAME}_osw_cygnss` from the variable changes? 

when I check the data in the `$JEDI_BUILD/iodaconv/test/testrun/obs...nc` file I see these values for the `ObsValue`:
```
    windEastward = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ; 

   windNorthward = 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  
      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ; 

   windSpeed = 3.124555, 12.34681, 4.752041, 8.625623, 6.620109, 5.329083, 
      4.058743, 12.31072, 10.10221, 3.397342, 2.156834, 9.655927, 3.10234, 
      12.34681, 8.625623, 4.752041, 6.620109, 5.329083, 3.802229, 11.92991, 
      3.397342, 11.08689, 2.156834, 9.722462, 12.34681, 8.625623, 3.10234, 
      4.830401, 6.16203, 5.329083, 2.156834, 3.397342, 3.637542, 11.92991, 
      12.29727, 9.722462, 2.803087, 12.34681, 8.682253, 4.965754, 5.329083, 
      5.697055, 11.86619, 3.628951, 3.560098, 12.92894, 9.739804, 2.722503, 
      2.156834, 4.978159, 8.682253, 4.904152, 5.194953, 11.86434, 3.613081, 
      3.831908, 13.80987, 1.862707, 9.739804, 2.717125, 13.31608, 5.16046, 
      8.610349, 4.309865, 5.194953, 11.47207, 3.522805, 1.862707, 14.02247, 
      4.150004, 9.776895, 2.609341, 12.84049, 5.618515, 8.488731, 5.145812, 
      3.959994, 4.248464 ;
```

**Comment by fabiolrdiniz** [comment]
> > Whenever testing, please remember to sync the filters of https://github.com/JCSDA-internal/skylab/blob/develop/obs/ufo/osw_muon.yaml with the ones for CYGNSS & Spire.
> 
> @fabiolrdiniz are you just saying we should update the skylab yaml for muon?

Yes, some of the filters were not added for Muon (error assignment, preqc, and background check).

**Review by haydenlj — APPROVED** [review]
Thanks @BenjaminRuston , I was able to run

**Comment by BenjaminRuston** [comment]
thanks for the reviews, and thanks for reminding us of the values @smaticka,

you've reported this didn't work if we set the values to missing.  So this is something we should record and either come back to, or complete the full operator.

**Pull request merged**